TESTS/mbedmicro-rtos-mbed/threads/main.cpp

```diff
@@ -108,37 +108,302 @@ void test_self_terminate() {
     delete thread;
 }
 
+void wait_set_signal(Thread *t) {
```

Comment:
Is it possible to split out cases and group together in a class various functions involved during the test case ?